core/src/main/java/de/danoeh/antennapod/core/storage/DBWriter.java

```diff
@@ -136,6 +130,20 @@ private DBWriter() {
         });
     }
 
+    /**
+     * Deletes a downloaded FeedMedia file from the storage device.
+     *
+     * @param context A context that is used for opening a database connection.
+     * @param mediaId ID of the FeedMedia object whose downloaded file should be deleted.
+     */
+    public static Future<?> deleteFeedMediaOfItem(final Context context,
+                                                  final long mediaId) {
+        if (UserPreferences.shouldDeleteRemoveFromQueue()) {
+            DBWriter.removeQueueItem(context, DBReader.getFeedMedia(mediaId).getItem(), false);
+        }
+        return doDeleteFeedMediaOfItem(context, mediaId);
```

Comment:
The main logic is run at the end: it is somewhat counter-intuitive. 
 
More of a style issue: as both the `dodeleteFeedMediaOfItem()` and `removeQueueItem()` are really run asynchronously anyway.